workspaces/ui-v2/src/components/sharing/ShareButton.tsx

```diff
@@ -0,0 +1,139 @@
+import React, { useCallback, useMemo, useState } from 'react';
+import {
+  Button,
+  Dialog,
+  DialogContent,
+  DialogActions,
+  makeStyles,
+  Typography,
+} from '@material-ui/core';
+import ShareIcon from '@material-ui/icons/Share';
+import { ShareModal } from './ShareModal';
+import { useAuth0 } from '@auth0/auth0-react';
+import { useSpecRepository } from '<src>/contexts/SpecRepositoryContext';
+import { useAppConfig } from '<src>/contexts/config/AppConfiguration';
+import { useAsyncMemo } from 'use-async-memo';
+
+export enum ShareTarget {
+  TEAM,
+  CUSTOMER,
+}
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    marginRight: theme.spacing(1),
+    height: 25,
+  },
+}));
+
+export const ShareButton: React.FC<{}> = (props) => {
+  const styles = useStyles(props);
+
+  const { getAccessTokenSilently, isAuthenticated, logout } = useAuth0();
+  const { specRepo } = useSpecRepository();
+  const {
+    api: { domain: baseDomain },
+    sharing,
+  } = useAppConfig();
+
+  const personId = useAsyncMemo(async () => {
+    if (isAuthenticated) {
+      const token = await getAccessTokenSilently();
+
+      const response = await fetch(`${baseDomain}/api/person`, {
+        headers: { Authorization: `Bearer ${token}` },
+      });
+      const resp_data = await response.json();
+      return resp_data.id;
+    }
+  }, [getAccessTokenSilently, isAuthenticated, baseDomain]);
```

Comment:
Can we put this state into redux? It might be good to put this into the `metadata` slice (we are slowly moving everything into redux - excluding non-serializable things like SpecRepo / spectacle, since it's likely we'll want to use the person's id / are they logged in, in other places in the app)

When / how does a user log in? Can they log in through the UI - if so, the relevant state will need to change if a user can go from not-logged in -> logged in

Reply:
Oh I see there's a login flow, we could load the initial user's state (in the inital metadata fetch) - and then as a callback for loading a user, it could hydrate the redux store)